### ISSUE TITLE
FR: list dependency of a shell script

### ISSUE THREAD

**Issue by glyh** [oilshell/oil#1436]
Say I'm writing some scripts to bootstrap my OS. I may need to know what this shell script depends on(what environment variables are used, what binaries are expected on path, etc. I think this should be possible to implement as a plugin, but at least we need to have API to make it possible.

**Comment by andychu** [comment]
Yes this is definitely something I've thought about and want to address, although I don't think there's a single clean solution

@abathur has written a tool on top of Oil that can do a bit of this

Although it necessarily has to be an approximation because names are generally dynamic

I guess what I've been saying is that changing the language makes this easier.  There is some incomplete code here to do

```
use bin grep sed
```

Although this design is not set in stone

https://github.com/oilshell/oil/blob/master/osh/builtin_pure.py#L637
